packages/nx/src/utils/package-manager.ts

```diff
@@ -29,19 +29,26 @@ export interface PackageManagerCommands {
   run: (script: string, args: string) => string;
 }
 
+/**
+ * Detects which package manager is used in the workspace based on the lock file.
+ */
+function getPackageManager(dir: string = '') {
+  return existsSync(join(dir, 'yarn.lock'))
+    ? 'yarn'
+    : existsSync(join(dir, 'bun.lockb'))
+    ? 'bun'
+    : existsSync(join(dir, 'pnpm-lock.yaml')) ||
+      existsSync(join(dir, 'pnpm-workspace.yaml'))
+    ? 'pnpm'
+    : 'npm';
+}
+
 /**
  * Detects which package manager is used in the workspace based on the lock file.
  */
 export function detectPackageManager(dir: string = ''): PackageManager {
   const nxJson = readNxJson();
-  return (
-    nxJson.cli?.packageManager ??
-    (existsSync(join(dir, 'yarn.lock'))
-      ? 'yarn'
-      : existsSync(join(dir, 'pnpm-lock.yaml'))
-      ? 'pnpm'
-      : 'npm')
-  );
+  return nxJson.cli?.packageManager ?? getPackageManager(dir);
 }
```

Comment:
Used util to call default lock file lookup

Reply:
Please revert this so that the logic is back to being inline in this small utility function. There is no value in splitting this out into its own function when the additional logic is 2 lines and the other function is private and never reused (see the comments above for the exact logic to use and comment to add)